Update user manual for scaladoc (#12083)
lampepfl/dotty: docs/docs/usage/scaladoc/staticSite.md

default table of contents allows you to have subsections - albeit the current
depth limit is 2 however it accepts both files and directories and latter can be used to provide deeper structures.

The items must provide either `subsection` or `url` but not both at once!

> The only excepiton is `Blog` which is only a `title` and behaves differently.

Reviewer: ```suggestion
The only exception is `Blog` which is only a `title` and behaves differently.
```

Author: Fixed

You can read more about blog [here](blog.md).

```
├── blog
│   ├── _posts
│   │   └── 2016-12-05-implicit-function-types.md
│   └── index.html
├── index.html
└── sidebar.yml
```

## Static resources

> You can attach static resources (pdf, images) in your documentation using two dedicated directories:

Reviewer: ```suggestion
You can attach static resources (pdf, images) to your documentation by using two dedicated directories:
```

Author: Fixed

`resources` and `images`. When you upload your assests under any of these directories you can reference them in markdown
as if they were relatively at the same level.

> For example let's look at this situation:

Reviewer: ```suggestion
For example, consider the following situation:
```

Author: Fixed

```
├── blog
│   ├── _posts
│   │   └── 2016-12-05-implicit-function-types.md
│   └── index.html
├── index.html
├── resources
│   └── my_file.pdf
├── images
│   └── my_image.png
└── sidebar.yml

```

> You can access them directly inside any of the files using markdown links:

Reviewer: ```suggestion
You can refer to the assets from within any of the files using markdown links:
```

Author: Fixed
